pandas/core/indexing.py

```diff
@@ -1821,7 +1821,7 @@ def _setitem_single_block(self, indexer, value, name: str):
                 return
 
             indexer = maybe_convert_ix(*indexer)
-        if isinstance(value, ABCSeries) and name != "iloc" or isinstance(value, dict):
+        if (isinstance(value, ABCSeries) and name != "iloc") or isinstance(value, dict):
```

Comment:
you said we have tests that get here with dict for both loc and iloc?

Reply:
yes, I have added ``test_setitem_iloc_dictionary_value`` for the iloc case.
``test_indexing_assignment_dict_already_exists`` covers the loc case

Reply:
thanks